### ISSUE TITLE
Compile error: "_FORTIFY_SOURCE" redefined [-Werror]

### ISSUE THREAD

**Issue by powerman** [majn/telegram-purple#385]
Last time I've successfully built it was about a month ago (Apr 2), but that time I've used gcc-4.9.4, while now I'm using gcc-5.4.0. So, either something change in the repo in last month, or this is an issue with newer gcc.
```
>>> Configuring source in /var/tmp/portage/x11-plugins/telegram-purple-9999/work/telegram-purple-9999 ...
./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --docdir=/usr/share/doc/telegram-purple-9999 --htmldir=/usr/share/doc/telegram-purple-9999/html --libdir=/usr/lib64 --enable-libwebp
checking for x86_64-pc-linux-gnu-gcc... x86_64-pc-linux-gnu-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether x86_64-pc-linux-gnu-gcc accepts -g... yes
checking for x86_64-pc-linux-gnu-gcc option to accept ISO C89... none needed
checking for a BSD-compatible install... /usr/lib/portage/python3.4/ebuild-helpers/xattr/install -c
checking for clock_gettime in -lrt... yes
checking for gcry_mpi_snatch in -lgcrypt... yes
checking how to run the C preprocessor... x86_64-pc-linux-gnu-gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking if zlib is wanted... yes
checking for inflateEnd in -lz... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for x86_64-pc-linux-gnu-pkg-config... /usr/bin/x86_64-pc-linux-gnu-pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for PURPLE... yes
checking for libwebp... enabled
checking for WebPDecodeRGBA in -lwebp... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking for unistd.h... (cached) yes
checking for size_t... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... yes
checking for memset... yes
checking for strdup... yes
checking for msgfmt... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
>>> Source configured.
>>> Compiling source in /var/tmp/portage/x11-plugins/telegram-purple-9999/work/telegram-purple-9999 ...
make -j8 
cd tgl && ./configure -q --disable-openssl --disable-extf CFLAGS="-march=native -O2 -pipe" LDFLAGS="-Wl,-O1 -Wl,--as-needed -L/usr/lib -L/usr/lib"
/usr/bin/msgfmt -cf -o po/bg.mo po/bg.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/cs_CZ.mo po/cs_CZ.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/de_DE.mo po/de_DE.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/es_AR.mo po/es_AR.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/fr.mo po/fr.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/it_IT.mo po/it_IT.po >/dev/null 2>&1
Refresh commit.h
/usr/bin/msgfmt -cf -o po/nl.mo po/nl.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/pl_PL.mo po/pl_PL.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/ru_RU.mo po/ru_RU.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/sq.mo po/sq.po >/dev/null 2>&1
/usr/bin/msgfmt -cf -o po/uk.mo po/uk.po >/dev/null 2>&1
make -C tgl libs/libtgl.a
make[1]: Entering directory '/var/tmp/portage/x11-plugins/telegram-purple-9999/work/telegram-purple-9999/tgl'
cat scheme.tl encrypted_scheme.tl binlog.tl mtproto.tl append.tl > auto/scheme.tl
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -iquote ./tl-parser -c -MP -MD -MF dep/tl-parser.d -MQ objs/tl-parser.o -o objs/tl-parser.o tl-parser/tl-parser.c
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -iquote ./tl-parser -c -MP -MD -MF dep/tlc.d -MQ objs/tlc.o -o objs/tlc.o tl-parser/tlc.c
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -c -MP -MD -MF dep/generate.d -MQ objs/generate.o -o objs/generate.o generate.c
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -c -MP -MD -MF dep/tools.d -MQ objs/tools.o -o objs/tools.o tools.c
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -c -MP -MD -MF dep/crypto/rand_openssl.d -MQ objs/crypto/rand_openssl.o -o objs/crypto/rand_openssl.o crypto/rand_openssl.c
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -c -MP -MD -MF dep/crypto/rand_altern.d -MQ objs/crypto/rand_altern.o -o objs/crypto/rand_altern.o crypto/rand_altern.c
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -c -MP -MD -MF dep/crypto/err_openssl.d -MQ objs/crypto/err_openssl.o -o objs/crypto/err_openssl.o crypto/err_openssl.c
gcc -I. -I. -march=native -O2 -pipe  -I/usr/local/include -I/usr/include -I/usr/include  -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong -c -MP -MD -MF dep/crypto/err_altern.d -MQ objs/crypto/err_altern.o -o objs/crypto/err_altern.o crypto/err_altern.c
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
cc1: all warnings being treated as errors
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
cc1: all warnings being treated as errors
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
make[1]: *** [Makefile:60: objs/crypto/rand_openssl.o] Error 1
make[1]: *** Waiting for unfinished jobs....
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
make[1]: *** [Makefile:60: objs/crypto/err_openssl.o] Error 1
cc1: all warnings being treated as errors
make[1]: *** [Makefile:60: objs/crypto/rand_altern.o] Error 1
cc1: all warnings being treated as errors
make[1]: *** [Makefile:60: objs/crypto/err_altern.o] Error 1
cc1: all warnings being treated as errors
make[1]: *** [Makefile.tl-parser:4: objs/tlc.o] Error 1
cc1: all warnings being treated as errors
make[1]: *** [Makefile:60: objs/tools.o] Error 1
cc1: all warnings being treated as errors
make[1]: *** [Makefile.tl-parser:4: objs/tl-parser.o] Error 1
cc1: all warnings being treated as errors
make[1]: *** [Makefile:60: objs/generate.o] Error 1
make[1]: Leaving directory '/var/tmp/portage/x11-plugins/telegram-purple-9999/work/telegram-purple-9999/tgl'
make: *** [Makefile:87: tgl/libs/libtgl.a] Error 2
```

**Comment by BenWiederhake** [comment]
Duplicate of #385, and fixed in branch `dev-1.4.0`.

The next version will take quite a while.  The relevant patch just removes `-Werror` everywhere, see #288.

**Comment by powerman** [comment]
You mean duplicate of #381.

**Comment by BenWiederhake** [comment]
Yeah, sorry.

**Comment by BenWiederhake** [comment]
Should be fixed in [v1.4.1](https://github.com/majn/telegram-purple/releases/tag/v1.4.1).

**Comment by powerman** [comment]
Yeah, I can confirm current master works with gcc-8.2.0, thanks!
